### ISSUE TITLE
Installing latest from master fails

### ISSUE THREAD

**Issue by rafspiny** [DaveMDS/python-efl#6]
Hi, I have been trying to install the latest code from python-efl, but I am failing to do so.
I am using the latest EFL from GIT. Not the latest code released.

Any idea what it could be?

> cython --version
> Cython version 3.0.2
✔ /tmp/tmp.egRQyrTG4H/python-efl [master|✔]
python --version
> Python 3.11.5

```
ARGET="lua5-1 -luajit" pulled in by
>>> Emerging (1 of 10) dev-python/python-efl-9999::enlightenment-live
 * Fetching files in the background.
 * To view fetch progress, run in another terminal:
 * tail -f /var/log/emerge-fetch.log
>>> Unpacking source...
Initialized empty Git repository in /var/cache/distfiles/git3-src/DaveMDS_python-efl.git/
 * Repository id: DaveMDS_python-efl.git
 * To override fetched repository properties, use:
 *   EGIT_OVERRIDE_REPO_DAVEMDS_PYTHON_EFL
 *   EGIT_OVERRIDE_BRANCH_DAVEMDS_PYTHON_EFL
 *   EGIT_OVERRIDE_COMMIT_DAVEMDS_PYTHON_EFL
 *   EGIT_OVERRIDE_COMMIT_DATE_DAVEMDS_PYTHON_EFL
 *
 * Fetching https://github.com/DaveMDS/python-efl.git ...
git fetch https://github.com/DaveMDS/python-efl.git +HEAD:refs/git-r3/HEAD
remote: Enumerating objects: 13249, done.
remote: Counting objects: 100% (289/289), done.
remote: Compressing objects: 100% (210/210), done.
remote: Total 13249 (delta 119), reused 122 (delta 77), pack-reused 12960
Receiving objects: 100% (13249/13249), 13.62 MiB | 3.71 MiB/s, done.
Resolving deltas: 100% (9994/9994), done.
From https://github.com/DaveMDS/python-efl
 * [new ref]         HEAD       -> refs/git-r3/HEAD
 * [new tag]         v1.10.0    -> v1.10.0
 * [new tag]         v1.11.0    -> v1.11.0
 * [new tag]         v1.12.0    -> v1.12.0
 * [new tag]         v1.13.0    -> v1.13.0
 * [new tag]         v1.14.0    -> v1.14.0
 * [new tag]         v1.15.0    -> v1.15.0
 * [new tag]         v1.16.0    -> v1.16.0
 * [new tag]         v1.17.0    -> v1.17.0
 * [new tag]         v1.18.0    -> v1.18.0
 * [new tag]         v1.19.0    -> v1.19.0
 * [new tag]         v1.20.0    -> v1.20.0
 * [new tag]         v1.21.0    -> v1.21.0
 * [new tag]         v1.22.0    -> v1.22.0
 * [new tag]         v1.23.0    -> v1.23.0
 * [new tag]         v1.24.0    -> v1.24.0
 * [new tag]         v1.25.0    -> v1.25.0
 * [new tag]         v1.26.0    -> v1.26.0
 * [new tag]         v1.8.0     -> v1.8.0
 * [new tag]         v1.9.0     -> v1.9.0
git symbolic-ref refs/git-r3/dev-python/python-efl/0/__main__ refs/git-r3/HEAD
 * Checking out https://github.com/DaveMDS/python-efl.git to /var/tmp/portage/dev-python/python-efl-9999/work/python-efl-9999 ...
git checkout --quiet refs/git-r3/HEAD
GIT NEW branch -->
   repository:               https://github.com/DaveMDS/python-efl.git
   at the commit:            a7464e66e61a52dcbd72f66f538f15d76d852465
>>> Source unpacked in /var/tmp/portage/dev-python/python-efl-9999/work
>>> Preparing source in /var/tmp/portage/dev-python/python-efl-9999/work/python-efl-9999 ...
 * Applying python-efl-1.25-clang-crosscompile.patch ...
patching file setup.py
Hunk #1 succeeded at 124 with fuzz 2 (offset -97 lines).                                                                                                                                                                       [ ok ]
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/dev-python/python-efl-9999/work/python-efl-9999 ...
>>> Source configured.
>>> Compiling source in /var/tmp/portage/dev-python/python-efl-9999/work/python-efl-9999 ...
 * python3_10: running distutils-r1_run_phase distutils-r1_python_compile
 *   Building the wheel for python-efl-9999 via setuptools.build_meta
gpep517 build-wheel --prefix=/usr --backend setuptools.build_meta --output-fd 3 --wheel-dir /var/tmp/portage/dev-python/python-efl-9999/work/python-efl-9999-python3_10/wheel
2023-11-01 15:50:54,339 gpep517 INFO Building wheel via backend setuptools.build_meta
warning: efl/dbus_mainloop/efl.dbus_mainloop.pyx:1:0: Dotted filenames ('efl.dbus_mainloop.pyx') are deprecated. Please use the normal Python package directory layout.
warning: efl/dbus_mainloop/efl.dbus_mainloop.pyx:1:0: Dotted filenames ('efl.dbus_mainloop.pxd') are deprecated. Please use the normal Python package directory layout.

Error compiling Cython file:
------------------------------------------------------------
...
cdef dbus_bool_t dbus_py_ecore_set_up_conn(DBusConnection *conn, void *data) with gil:
    e_dbus_connection_setup(conn)
    return True
cdef object dbus_ecore_native_mainloop():
    return <object>DBusPyNativeMainLoop_New4(dbus_py_ecore_set_up_conn,
                                             ^
------------------------------------------------------------

efl/dbus_mainloop/efl.dbus_mainloop.pyx:33:45: Cannot assign type 'dbus_bool_t (DBusConnection *, void *) except? -1 nogil' to '_dbus_py_conn_setup_func'. Exception values are incompatible. Suggest adding 'noexcept' to type 'dbus_bool_t (DBusConnection *, void *) except? -1 nogil'.
Python-EFL: 1.26.99a1258
Checking for Python: OK, found 3.10.13
Checking for Cython: OK, found 3.0.2
Checking for Eina: OK, found 1.26.99
Checking for Eo: OK, found 1.26.99
Checking for Evas: OK, found 1.26.99
Checking for Ecore: OK, found 1.26.99
Checking for EcoreFile: OK, found 1.26.99
Checking for EcoreInput: OK, found 1.26.99
Checking for EcoreCon: OK, found 1.26.99
Checking for EcoreX: OK, found 1.26.99
Checking for Ethumb: OK, found 1.26.99
Checking for Ethumb_Client: OK, found 1.26.99
Checking for Edje: OK, found 1.26.99
Checking for Emotion: OK, found 1.26.99
Checking for DBus: OK, found 1.3.2
Checking for Elementary: OK, found 1.26.99
Compiling efl/eo/efl.eo.pyx because it changed.
Compiling efl/utils/deprecated.pyx because it changed.
Compiling efl/utils/conversions.pyx because it changed.
Compiling efl/utils/logger.pyx because it changed.
Compiling efl/evas/efl.evas.pyx because it changed.
Compiling efl/ecore/efl.ecore.pyx because it changed.
Compiling efl/ecore_input/efl.ecore_input.pyx because it changed.
Compiling efl/ecore_con/efl.ecore_con.pyx because it changed.
Compiling efl/ecore_x/efl.ecore_x.pyx because it changed.
Compiling efl/ethumb/efl.ethumb.pyx because it changed.
Compiling efl/ethumb/efl.ethumb_client.pyx because it changed.
Compiling efl/edje/efl.edje.pyx because it changed.
Compiling efl/edje_edit/efl.edje_edit.pyx because it changed.
Compiling efl/emotion/efl.emotion.pyx because it changed.
Compiling efl/dbus_mainloop/efl.dbus_mainloop.pyx because it changed.
Compiling efl/elementary/__init__.pyx because it changed.
[ 1/16] Cythonizing efl/dbus_mainloop/efl.dbus_mainloop.pyx
Traceback (most recent call last):
  File "/usr/lib/python-exec/python3.10/gpep517", line 4, in <module>
    sys.exit(main())
  File "/usr/lib/python3.10/site-packages/gpep517/__main__.py", line 425, in main
    return func(args)
  File "/usr/lib/python3.10/site-packages/gpep517/__main__.py", line 215, in build_wheel
    print(build_wheel_impl(args, args.wheel_dir), file=out)
  File "/usr/lib/python3.10/site-packages/gpep517/__main__.py", line 207, in build_wheel_impl
    wheel_name = backend.build_wheel(str(wheel_dir), args.config_json)
  File "/usr/lib/python3.10/site-packages/setuptools/build_meta.py", line 434, in build_wheel
    return self._build_with_temp_dir(
  File "/usr/lib/python3.10/site-packages/setuptools/build_meta.py", line 419, in _build_with_temp_dir
    self.run_setup()
  File "/usr/lib/python3.10/site-packages/setuptools/build_meta.py", line 341, in run_setup
    exec(code, locals())
  File "<string>", line 307, in <module>
  File "/usr/lib/python3.10/site-packages/Cython/Build/Dependencies.py", line 1139, in cythonize
    cythonize_one(*args)
  File "/usr/lib/python3.10/site-packages/Cython/Build/Dependencies.py", line 1306, in cythonize_one
    raise CompileError(None, pyx_file)
Cython.Compiler.Errors.CompileError: efl/dbus_mainloop/efl.dbus_mainloop.pyx


```

**Comment by TheTechRobo** [comment]
What is the installed version of Cython?

**Comment by rafspiny** [comment]
@TheTechRobo it is in my original message: Cython version 3.0.2

**Comment by TheTechRobo** [comment]
Ah, apologies - I missed that. Try installing Cython `0.29.36`.

a7464e66e61a52dcbd72f66f538f15d76d852465 was supposed to fix the build, but it only enforces the minimum version, not the maximum version - I think Cython 3.x is incompatible.

**Comment by DaveMDS** [comment]
Yes, you need cython 2.X ti build from source,

Or Just install from pip

**Comment by rafspiny** [comment]
Oh thanks @TheTechRobo .

I made a quick fix, if it is acceptable. #7

**Comment by DaveMDS** [comment]
Fixed in c8d7bdd
